test/gif-encoder_test.js

```diff
@@ -150,3 +150,42 @@ describe('GifEncoder encoding a multi-framed image with a transparent background
     assert.deepEqual(expectedBytes, actualBytes);
   });
 });
+
+describe('GifEncoder encoding a checkerboard from indexed pixels', function () {
+  createGif(10, 10, {
+    palette: [0, 0, 0, 255, 255, 255]
+  });
+  const sz = 10;
+  const indexedPixels = Array(sz*sz).fill(0);
+  for (let y = 0; y < sz; y++) {
+    for (let x = 0; x < sz; x++) {
+      const offs = y*sz + x;
+      const white = Math.floor(y/(sz/2)) ^ Math.floor(x/(sz/2)) ^ 1;
```

Comment:
Can you explain this line further?

Reply:
This creates a 10x10 checkerboard pattern (5x5 square) of indexed colors 0 and 1. It was easier to generate programmatically than author a .gif for it.

If you run in the debug mode you can see the gif it outputs. 

The xor trick is how one usually generates a 2d checkerboard pattern, in case that’s the unclear part.  

The variable name ”white” is probably not the greatest choice. Maybe just ”colorIndex” would be better.

Reply:
This creates a 10x10 checkerboard pattern (5x5 square) of indexed colors 0 and 1. It was easier to generate programmatically than author a .gif for it.

If you run in the debug mode you can see the gif it outputs. 

The xor trick is how one usually generates a 2d checkerboard pattern, in case that’s the unclear part.  

The variable name ”white” is probably not the greatest choice. Maybe just ”colorIndex” would be better.

Reply:
Ah, I see. It might be even better to enumerate the index values as an array literal -- that would avoid future confusion as well:

```js
var indexedPixels = [
  0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
  1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
  0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
  1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
  0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
  1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
  0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
  1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
  0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
  1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
];
```